### PR TITLE
JDK-8305673: Convert DocCommentParser to use enhanced switch

### PR DESCRIPTION
**Pull request by jonathan-gibbons**
Please review more cleanup for DocCommentParser, primarily focused on using enhanced switch statements and expressions.

The first commit is mostly IDE-automated.
The second commit is manual.

All change are "local".  There are two remaining switch statements using the old form that have non-trivial fall-through code, which could/should be moved out into separate methods.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8305673](https://bugs.openjdk.org/browse/JDK-8305673): Convert DocCommentParser to use enhanced switch

### Reviewers
 * [Hannes Wallnöfer](https://openjdk.org/census#hannesw) (@hns - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/13362/head:pull/13362` \
`$ git checkout pull/13362`

Update a local copy of the PR: \
`$ git checkout pull/13362` \
`$ git pull https://git.openjdk.org/jdk.git pull/13362/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 13362`

View PR using the GUI difftool: \
`$ git pr show -t 13362`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/13362.diff">https://git.openjdk.org/jdk/pull/13362.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/13362#issuecomment-1498158314)

### CODE REVIEW

File: src/jdk.compiler/share/classes/com/sun/tools/javac/parser/DocCommentParser.java [hunk 1]
```diff
@@ -593,6 +593,7 @@ private List<DCTree> inlineContent() {
         int depth = 1;
         textStart = -1;
 
+        loop:
```
Comment (hns) [inline src/jdk.compiler/share/classes/com/sun/tools/javac/parser/DocCommentParser.java]: This label doesn't seem to be used, why was it added back?

**Review by hns — APPROVED** [review]
Looks good to me, but the 4th commit adding back the `loop` label doesn't seem to be necessary.

File: src/jdk.compiler/share/classes/com/sun/tools/javac/parser/DocCommentParser.java (hunk 1 above)
Comment (jonathan-gibbons) [reply src/jdk.compiler/share/classes/com/sun/tools/javac/parser/DocCommentParser.java]: It was just to try and minimize changes with downstream commits, such as PR #13431, and didn't seem important to clean up in this round.

**Pull request closed without merging**